pkg/notifier/notifier_chain.go

```diff
@@ -0,0 +1,23 @@
+package notifier
+
+type NotifierChain struct {
+	notifierProvider NotifierProvider
+}
+
+func NewNotifierChain(notifierProvider *NotifierProvider) *NotifierChain {
+	return &NotifierChain{notifierProvider: *notifierProvider}
```

Comment:
`NotifierChain` structs expect `NotiferProvider`, we give `*NotifierProvider` to `NewNotifierChain`, which translates it back into `NotifierProvider`? How about not using pointer at all and just use some interface with `GetNotifiers()` function and write unit test for it?